setup.cfg

```diff
@@ -1,3 +1,58 @@
+[metadata]
```

Comment:
(Disclaimer: I'm still new to `pyproject.toml`) Is there a reason to move this into `setup.cfg` and not `pyproject.toml`? It looks like other projects in the space, like [`pandas`](https://github.com/pandas-dev/pandas/blob/9b4cffca38cdc77ee83213af521a02508eff0cb1/pyproject.toml) and [`zarr`](https://github.com/zarr-developers/zarr-python/blob/55a875d16a7cb9f67dd652a5e921d5fe4bc37ac0/pyproject.toml),  have this in `pyproject.toml`.

Reply:
I'm going to run this through ini2toml but I'm waiting on https://github.com/abravalheri/ini2toml/issues/71

Reply:
Do we want to keep the `requirements.txt` file after making this move? FWIW in Dask-CUDA we decided to inline `requirements.txt` into `pyproject.toml` as part of the transition ( https://github.com/rapidsai/dask-cuda/pull/1035#discussion_r1013788761 )

Reply:
Yeah I'll inline it